browser/ui/split_view/split_view_link_redirect_utils.cc

```diff
@@ -0,0 +1,248 @@
+        tabs::TabInterface::MaybeGetFromContents(source);
+    if (!source_tab) {
+      return nullptr;
+    }
+
+    TabStripModel* tab_strip_model =
+        source_tab->GetBrowserWindowInterface()->GetTabStripModel();
+    if (!tab_strip_model) {
+      return nullptr;
+    }
+
+    split_tabs::SplitTabData* split_data =
+        tab_strip_model->GetSplitData(split_tab_id.value());
+    if (!split_data || !split_data->linked()) {
+      return nullptr;
+    }
+
+    // Get the right pane (second tab in the split)
+    const std::vector<tabs::TabInterface*> tabs_in_split =
+        split_data->ListTabs();
+    if (tabs_in_split.size() != 2) {
+      return nullptr;
+    }
+
+    return tabs_in_split[1]->GetContents();
+  }
+
+  if (out_from_window_open) {
+    *out_from_window_open = false;
+  }
+
+  // Original logic: check if source is already in a split view
+  tabs::TabInterface* source_tab =
+      tabs::TabInterface::MaybeGetFromContents(source);
+  if (!source_tab || !source_tab->IsSplit()) {
+    return nullptr;
+  }
+
+  // Get the parent collection and cast to SplitTabCollection
+  const tabs::TabCollection* parent_collection =
+      source_tab->GetParentCollection();
+  if (!parent_collection) {
+    return nullptr;
+  }
+
+  const tabs::SplitTabCollection* split_collection =
+      static_cast<const tabs::SplitTabCollection*>(parent_collection);
+  split_tabs::SplitTabData* split_data = split_collection->data();
+  if (!split_data || !split_data->linked()) {
+    return nullptr;
+  }
+
+  // Get the list of tabs in the split and verify this is the left pane
+  const std::vector<tabs::TabInterface*> tabs_in_split = split_data->ListTabs();
+  if (tabs_in_split.size() != 2 || tabs_in_split[0]->GetContents() != source) {
+    return nullptr;
+  }
+
+  return tabs_in_split[1]->GetContents();
+}
+
+}  // namespace
+
+namespace split_view {
+
+bool MaybeRedirectToRightPane(content::WebContents* source,
+                              const GURL& url,
+                              const content::Referrer& referrer) {
+  if (!base::FeatureList::IsEnabled(features::kSideBySide)) {
+    return false;
+  }
+
+  if (!source) {
+    return false;
+  }
+
+  bool from_window_open = false;
+  content::WebContents* target_contents =
+      GetRightPaneIfLinked(source, &from_window_open);
+  if (!target_contents) {
+    return false;
+  }
+
+  // We can load url via target contents' NavigationController
+  // but Navigate() is definitely the better choice as this is
+  // another browser initiated navigation.
+  tabs::TabInterface* target_tab =
+      tabs::TabInterface::GetFromContents(target_contents);
+  NavigateParams params(target_tab->GetBrowserWindowInterface(), url,
+                        ui::PAGE_TRANSITION_LINK);
+  params.source_contents = target_contents;
+  params.disposition = WindowOpenDisposition::CURRENT_TAB;
+  params.referrer = referrer;
+
+  // Don't want to set opener when routed from window.open().
```

Comment:
`window.open()` does set the `opener` unless the [`noopener` window feature](https://developer.mozilla.org/en-US/docs/Web/API/Window/open#noopener) is used. See https://fmarier.github.io/brave-testing/noopener.html for the difference in behavior between `<a>` and `window.open()`.

If we make it `null`, then we may break workflows that require the opener such has some SSO providers.